File: iosApp/iosApp/navigation/AppShellLayout.swift
import CoreGraphics

// 셸이 고를 수 있는 내비게이션 배치 — Compose navigation/AppShellLayout.kt 미러 (규칙 §14)
enum AppShellLayout {
    case bottomBar
    case rail
}

// 가로폭 하나로 배치를 정한다. 판정은 여기 한 곳에만 둔다
enum AppShellLayoutPolicy {
    // Material3 window size class의 compact/medium 경계
    static let railMinWidth: CGFloat = 600

    // 전 화면이 모바일 폭 기준 시안이라, 넓은 창에서는 본문을 이 폭으로 묶는다
    static let contentMaxWidth: CGFloat = 600

    // horizontalSizeClass가 아니라 실제 폭으로 잰다 — size class는 iPad 분할 화면에서
    // 경계가 600pt와 어긋나 3플랫폼 판정이 달라진다 (스펙 2026-09-16 §1-2)
    static func layoutFor(width: CGFloat) -> AppShellLayout {
        return width >= railMinWidth ? .rail : .bottomBar
    }
}

File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/navigation/AppShellLayout.kt
package org.sesacteamproject.passmate.navigation

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// 셸이 고를 수 있는 내비게이션 배치 (스펙 2026-09-16 §1-1)
enum class AppShellLayout {
    BOTTOM_BAR,
    RAIL
}

// 가로폭 하나로 배치를 정한다. 판정은 여기 한 곳에만 둔다 (규칙 §2-1-1의 barOwnerOf와 같은 원칙)
object AppShellLayoutPolicy {

    // Material3 window size class의 compact/medium 경계
    val RAIL_MIN_WIDTH: Dp = 600.dp

    // 재는 값은 셸이 받은 **창 폭**이다. 레일을 뺀 본문 폭으로 재면 600 근처에서 배치가 튄다
    // (레일이 붙으면 본문이 600 밑으로 내려가 다시 하단바가 되고, 그러면 다시 600을 넘는다)
    fun layoutFor(width: Dp): AppShellLayout {
        return if (width >= RAIL_MIN_WIDTH) AppShellLayout.RAIL else AppShellLayout.BOTTOM_BAR
    }
}
